src/Plutarch/Extra/StateThread.hs

```diff
@@ -59,7 +62,13 @@ uniqueStateTokenMint ::
   Term s (PValue keys amounts) ->
   Term s PBool
 uniqueStateTokenMint thisPolicy mint =
-  pvalueOf # mint # thisPolicy # pconstant "" #== pconstant 1
+  pisSingleton
+    # pto (pfromJust #$ plookup # thisPolicy # pto mint)
```

Comment:
Is it okay for this function to throw `perror`?

Reply:
It may look ugly but failing is the intended behavior.

Reply:
Hmm, okay. Since it's returning bool, I thought it shouldn't be partial.